pennylane/ops/functions/dot.py

```diff
@@ -136,6 +136,12 @@ def dot(
     if len(coeffs) == 0 and len(ops) == 0:
         raise ValueError("Cannot compute the dot product of an empty sequence.")
 
+    for t in (Operator, PauliWord, PauliSentence):
```

Comment:
Adjacent comment, but our type hint just says `sequence[Operator]`, should we update it to also include `PauliWord` and `PauliSentence`?

Reply:
yeah probably forgot to put that when I updated dot to accept Paulis

Reply:
added to docstring.